### ISSUE TITLE
Consider Verifiable Presentation Request design patterns

### ISSUE THREAD

**Issue by msporny** [w3c-fedid/digital-credentials#12]
Based on commentary in this issue about the complexity in PeX and it being viewed as too complicated for a presentation request language (I agree):

https://bitbucket.org/openid/connect/issues/1917/moving-to-a-credential-format-specific

The charter should consider the design approach taken in the Credential Handler API (CHAPI) and the Verifiable Presentation Request (VPR) work. Namely, that the design is meant to be protocol and query language agnostic. That is, it is designed to support mdoc query languages, VC query languages, and different protocols for mdoc, VCs, and other market vertical formats:

Demonstration of being query language agnostic:
https://w3c-ccg.github.io/vp-request-spec/#query-and-response-types

Demonstration of being protocol agnostic:
https://w3c-ccg.github.io/vp-request-spec/#interaction-types

While the examples are out of date, the design pattern still holds (and it looks like the discussion in the OpenID community is converging on the same design pattern that was established by VPR many years ago).

Fundamentally, the query languages and protocols for moving the newest types of digital credentials over the Web, NFC, and QR Codes are very new and unsettled. Trying to lock one in at this moment in time, or in a WG over the next 1-2 years is premature. Thus, the safest design approach would be to allow for agility at both the query language layer and the protocol layer.

**Comment by samuelgoto** [comment]
> The charter should consider the design approach taken in the Credential Handler API (CHAPI) and the Verifiable Presentation Request (VPR) work. Namely, that the design is meant to be protocol and query language agnostic. That is, it is designed to support mdoc query languages, VC query languages, and different protocols for mdoc, VCs, and other market vertical formats:

Yeah, we are really struggling with this question and we could really use your help and experience figuring this one out.

Can you help us by making a concrete proposal on what that API may look like here? I'm somewhat familiar with CHAPI ([as you know](https://github.com/fedidcg/FedCM/issues/374)), but I could use a concrete proposal to help me visualize in concrete terms what things would look like (I think I generally agree with the sense/direction you are proposing).

Could you help us re-write this [mDocs example](https://github.com/agl/identity-credential#mdocs) and this [VC example](https://github.com/agl/identity-credential#w3c-verifiable-credentials) in terms of what you are suggesting with [Verifiable Presentations](https://w3c-ccg.github.io/vp-request-spec/#query-and-response-types)?

Here is what I think it would look like, but can you check if this is what you are suggesting?

```javascript
// Gets a CBOR with specific fields out of mobile driver's license as an mdoc
const {response} = await navigator.credentials.get({
  identity: {
    providers: [{
      vc: {
        query: {
            // newly introduced query type
            type: "QueryByDocTypeAndElements",
            documentType: "org.iso.18013.5.1.mDL",
            requestedElements: [
              { namespace: "org.iso.18013.5.1", name: "document_number" },
              { namespace: "org.iso.18013.5.1", name: "portrait" },
              { namespace: "org.iso.18013.5.1", name: "driving_privileges" },
              { namespace: "org.iso.18013.5.1.aamva", name: "organ_donor" },
            ],
        }

        // Query agnostic / Protocol layer
        challenge: "gf69kepV+m5tGxUIsFtLi6pwg=",
        recipient: ["ftl+VEHPB17r2 ... Nioc9QZ7X/6w..."],
        // where should retention go?
        // retentionDays: 90,
      }
    }],
  }
});
```

Is this more or less what you have in mind?

**Comment by marcoscaceres** [comment]
Closing as duplicate of https://github.com/w3c-fedid/digital-credentials/issues/38 ... please let me know if you disagree.
